biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -286,7 +286,8 @@ public File get(String bsn, Version version, Map<String, String> properties, Dow
 		} else {
 
 			String name = resource.getInfo()
-				.name();
+				.name() + ":" + version;
```

Comment:
I agree that before this PR `resource.getInfo().name()` set by `BndPomRepository` also holds the version.

But as in your last comment described the MavenBndRepository uses as name `String name = archive.getWithoutVersion();` 

https://github.com/bndtools/bnd/blob/400cd7557624ba4428f074a9b3e0023ec220108c/biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java#L328-L334

so i changed to this behavior an run into exceptions because of the missing version. that is why i added the concatenatioC.

Could you please tell me which unit-test you had debugged?